src/flows/pipes/Schedule/ExportTaskOverlay/context.tsx

```diff
@@ -0,0 +1,209 @@
+import {
+  resourceLimitReached,
+  taskCreatedSuccess,
+  taskNotCreated,
+  taskUpdateSuccess,
+  taskUpdateFailed,
+} from 'src/shared/copy/notifications'
+import {ASSET_LIMIT_ERROR_STATUS} from 'src/cloud/constants/index'
+import {postTask, patchTask} from 'src/client'
+
+// Types
+import {Task} from 'src/types'
+
+// Actions
+import {getTasks} from 'src/tasks/actions/thunks'
+import {notify} from 'src/shared/actions/notifications'
+import {checkTaskLimits} from 'src/cloud/actions/limits'
+
+// Utils
+import {event} from 'src/cloud/utils/reporting'
+import {getErrorMessage} from 'src/utils/api'
+import {getOrg} from 'src/organizations/selectors'
+import {remove} from 'src/flows/context/query'
+
+// Contexts
+import {PopupContext} from 'src/flows/context/popup'
+
+export enum ExportAsTask {
+  Create = 'create',
+  Update = 'update',
+}
+
+interface ContextType {
+  script: string
+  submit: () => void
+  activeTab: ExportAsTask
+  handleSetActiveTab: (tab: ExportAsTask) => void
+  selectedTask: Task | undefined
+  handleSetTask: (task: Task) => void
+  taskName: string
+  handleSetTaskName: (name: string) => void
+}
+
+const DEFAULT_CONTEXT: ContextType = {
+  script: '',
+  submit: () => {},
+  activeTab: ExportAsTask.Create,
+  handleSetActiveTab: _ => {},
+  selectedTask: undefined,
+  handleSetTask: _ => {},
+  taskName: '',
+  handleSetTaskName: _ => {},
+}
+
+export const Context = React.createContext<ContextType>(DEFAULT_CONTEXT)
+
+export const Provider: FC = ({children}) => {
+  const dispatch = useDispatch()
+  const org = useSelector(getOrg)
+  const [activeTab, setActiveTab] = useState(ExportAsTask.Create)
+  const [selectedTask, setSelectedTask] = useState<Task>(undefined)
+  const [query, setQuery] = useState<string>('')
+  const [taskName, setTaskName] = useState<string>('')
+  const {data, closeFn} = useContext(PopupContext)
+
+  const _taskName =
+    activeTab === ExportAsTask.Create ? taskName : selectedTask?.name
+  const script = query
+
+  useEffect(() => {
+    const ast = parse(data.query)
+    const params = remove(
+      ast,
+      node =>
+        node.type === 'OptionStatement' && node.assignment.id.name === 'task'
+    ).reduce((acc, curr) => {
+      curr.assignment.init.properties.reduce((_acc, _curr) => {
+        _acc[_curr.key.name] = _curr.value.location.source
+        return _acc
+      }, acc)
+
+      return acc
+    }, {})
+
+    if (!_taskName) {
+      if (activeTab === ExportAsTask.Create) {
+        setTaskName(params.name.replace(/^"(.*)"$/, '$1') || 'Name this Task')
+      }
+    } else {
+      params.name = `"${_taskName || 'Name this task'}"`
+    }
+
+    const paramString = Object.entries(params)
+      .map(([key, val]) => `${key}: ${val}`)
+      .join(',\n')
+    const header = parse(`option task = {${paramString}}\n`)
+    ast.body.unshift(header.body[0])
+
+    setQuery(format_from_js_file(ast))
+  }, [_taskName, data.query])
+
+  useEffect(() => {
+    dispatch(getTasks())
```

Comment:
didn't you hate this?

Reply:
not this one. i was raging about all of the other update / add thunks that spiral out of control